### PR TITLE
Fix variation names

### LINKED ISSUE #3146: Variations: Open attributes are not displayed in the variation name
When some of the attributes are “open” in a variation, they're not part of the name. For example, the variations “Blue – Any material” and “Green – Any material” show up as “Blue” and “Green”.

### PR DESCRIPTION
**Pull request by 0nko**
Fixes #3146. This PR adds "Any _[attribute name]_" to the variation name for any open attribute of the variation.

**To test:**
1. Create variations with some or no options selected
2. Go to Products -> Variable product -> Variations
3. Verify the attribute choices are reflected in the variation names
4. Select an individual variation
5. Verify the attribute choices are reflected in the variation name in the toolbar

image

image

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/model/ProductVariation.kt [hunk 1]
```diff
@@ -162,7 +175,9 @@ fun WCProductVariationModel.toAppModel(): ProductVariation {
         ProductStockStatus.fromString(this.stockStatus),
         ProductBackorderStatus.fromString(this.backorders),
         this.stockQuantity,
-        getAttributeOptionName(this.getProductVariantOptions()),
+        this.getProductVariantOptions()
+            .filter { it.option != null && it.option != null }
```
Comment (anitaa1990) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/model/ProductVariation.kt]: Should this be `it.name != null && it.option != null`?

**Review by anitaa1990 — COMMENTED** [review]
Looks good! Just had a small doubt in the code. Once that is addressed, we can :shipit:

File: WooCommerce/src/main/kotlin/com/woocommerce/android/model/ProductVariation.kt (hunk 1 above)
Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/model/ProductVariation.kt]: Yup! :)

**Review by anitaa1990 — APPROVED** [review]
:shipit:

**Pull request merged**